src/utils/message.js

```diff
@@ -20,6 +20,10 @@ export const shouldBeMuted = (
     return false; // never hide a message when narrowed to topic
   }
 
+  if (isMentionedNarrow(narrow)) {
+    return false;
+  }
```

Comment:
I see the other early returns each have a comment explaining the reasoning. Could we add one that makes this one clearer (in particular, that this is just a tactical fix for now)?

Reply:
Ah, yeah. I don't really like the other comments (they're explaining *what* the code's doing, not *why*), but this is a case where explaining why would be useful. I'll add that.

Reply:
Yeah, exactly.

Reply:
The new comment is good!

I agree about the existing comments. The one on "when narrowed to topic" is especially unhelpful. I think a proper fix for that comment (and that conditional certainly does need some kind of comment) would involve writing some jsdoc to articulate what this function's semantics is supposed to be in the first place.

… And then where *that* leads, I think, is that this is really a helper function specialized for its one caller, `getShownMessagesForNarrow`, and should appear next to that function. (That is: it's hard to coherently write down what its meaning is supposed to be without effectively saying "this says whether `getShownMessagesForNarrow` should filter this message out".) And in turn looking at that call site… it should really get inlined there, because the top level of its control flow is mostly about switching on the type of narrow, and in many cases we can skip the `filter` entirely.

Anyway, none of that needs to happen in this PR. Perhaps after this is merged, I'll go and try cleaning that up and writing associated jsdoc.